examples/morpheusvm/tests/e2e/e2e_test.go

```diff
@@ -46,22 +66,141 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
+			Timeout:      20 * time.Second,
+		},
+		hload.DefaultGradualLoadOrchestratorConfig(),
+	)
+
+	return fixture.NewTestEnvironment(e2e.NewTestContext(), flagVars, owner, testingNetworkConfig, consts.ID).Marshal()
 }, func(envBytes []byte) {
 	// Run in every ginkgo process
 
 	// Initialize the local test environment from the global state
 	e2e.InitSharedTestEnvironment(ginkgo.GinkgoT(), envBytes)
+
+	tc := e2e.NewTestContext()
+	r := require.New(tc)
+
+	// Start metrics server
+	mux := http.NewServeMux()
+	mux.Handle("/ext/metrics", promhttp.HandlerFor(registry, promhttp.HandlerOpts{
+		Registry: registry,
+	}))
+
+	metricsServer := &http.Server{
+		Addr:              metricsURI,
+		Handler:           mux,
+		ReadHeaderTimeout: 10 * time.Second,
+	}
+
+	go func() {
+		r.ErrorIs(
+			metricsServer.ListenAndServe(),
+			http.ErrServerClosed,
+		)
+	}()
+
+	// Generate collector config
+	collectorConfigBytes, err := generateCollectorConfig(
+		[]string{metricsURI},
+		e2e.GetEnv(tc).GetNetwork().UUID,
+	)
+	r.NoError(err)
+
+	r.NoError(writeCollectorConfig(collectorConfigBytes))
+
+	ginkgo.DeferCleanup(func() {
+		homeDir, err := os.UserHomeDir()
+		r.NoError(err)
+		r.NoError(os.Remove(filepath.Join(homeDir, metricsFilePath)))
+		r.NoError(metricsServer.Shutdown(context.Background()))
+	})
 })
+
+func loadTxGenerators(
+	ctx context.Context,
+	uri string,
+	authFactories []chain.AuthFactory,
+) ([]hload.TxGenerator[*chain.Transaction], error) {
+	lcli := vm.NewJSONRPCClient(uri)
+	ruleFactory, err := lcli.GetRuleFactory(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	numOfFactories := len(authFactories)
+	balances := make([]uint64, numOfFactories)
+	// Get balances
+	for i, factory := range authFactories {
+		balance, err := lcli.Balance(ctx, factory.Address())
+		if err != nil {
+			return nil, err
+		}
+		balances[i] = balance
+	}
+
+	cli := jsonrpc.NewJSONRPCClient(uri)
+	unitPrices, err := cli.UnitPrices(ctx, false)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create tx generator
+	txGenerators := make([]hload.TxGenerator[*chain.Transaction], numOfFactories)
+	for i := 0; i < numOfFactories; i++ {
+		txGenerators[i] = load.NewTxGenerator(authFactories[i], ruleFactory, balances[i], unitPrices)
+	}
+
+	return txGenerators, nil
+}
+
+func generateCollectorConfig(targets []string, uuid string) ([]byte, error) {
+	nodeLabels := tmpnet.FlagsMap{
+		"network_owner": "hypersdk-e2e-tests",
+		"network_uuid":  uuid,
+	}
+	config := []tmpnet.FlagsMap{
+		{
+			"labels":  nodeLabels,
+			"targets": targets,
+		},
+	}
+
+	return json.Marshal(config)
+}
+
+func writeCollectorConfig(config []byte) error {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+	fmt.Println(homeDir)
+	file, err := os.OpenFile(filepath.Join(homeDir, metricsFilePath), os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0o644)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	if _, err := file.Write(config); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
Is this specific to MorpheusVM? If not, would it make sense to move to `e2e.go` so that other VMs do not need to re-implement?